src/Catalog/Helpers/NonhijackableV2HttpMessageHandlerWrapper.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// A <see cref="HttpMessageHandler"/> that wraps another <see cref="HttpMessageHandler"/> and prevents queries to the V2 feed from being hijacked by the search service.
+    /// </summary>
+    public class NonhijackableV2HttpMessageHandlerWrapper : HttpMessageHandler
+    {
+        /// <param name="inner">The <see cref="HttpMessageHandler"/> to wrap.</param>
+        public NonhijackableV2HttpMessageHandlerWrapper(HttpMessageHandler inner)
+        {
+            _innerHttpMessageHandler = inner ?? throw new ArgumentNullException(nameof(inner));
+        }
+
+        private HttpMessageHandler _innerHttpMessageHandler;
+
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            request.RequestUri = GetNonhijackableUri(request.RequestUri);
+
+            /// For whatever reason, <see cref="HttpMessageHandler.SendAsync"/> is a protected method.
+            /// As a result, we must use reflection to call <see cref="HttpMessageHandler.SendAsync"/> on <see cref="_innerHttpMessageHandler"/>.
+            var sendAsyncMethod = _innerHttpMessageHandler.GetType().GetMethod(nameof(SendAsync), BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
>GetMethod [](start = 69, length = 9)

Reflection is the wrong way. Use DelegatingHandler. Before you start using reflection, you should exhaust investigation of other approaches.